floodviz/peak_flow_utils.py

```diff
@@ -1,104 +1,125 @@
     """
     # peak value historical data #
-    content = None
-    url = url_prefix + '?site_no=' + site + '&agency_cd=USGS&format=rdb' + '&end_date=' + end_date
-    try:
-        r = requests.get(url)
-    except requests.exceptions.RequestException as e:
-        print('- Bad URL -')
+    params = {
+        'agency_cd': 'USGS',
+        'format': 'rdb',
+        'site_no': site,
+        'end_date': end_date
+    }
+    raw_peaks = utils.parse_rdb(url_prefix, params)
+    if raw_peaks is None:
+        print('rdb parser has returned no data to peakflow util req_peak_data')
+        peaks = None
+
     else:
-        if r.status_code is 200:
-            content = r.text.splitlines()
-    
-    return content
+        keep_keys = [
+            'peak_dt',
+            'peak_va'
+        ]
+        peaks = []
+        for raw in raw_peaks:
+            peak = {k: raw[k] for k in keep_keys}
+            peaks.append(peak)
 
-def req_peak_dv_data(site, date, url_prefix):
-    """ 
-    requests data from the Daily Value NWIS water data service
-    for creating the lollipop svg elements for the current year.
+    return peaks
 
-    ARGS: 
-        site - string site ID for the site to be charted
-        date - date to chart lollipop flow data
-        url_pefix - String constant for nwis peak waterdata service endpoint url
-    
-    RETURNS:
-        content - list of all lines in the data file 
 
+def req_peak_dv_data(site, date, url_prefix):
     """
-    content = None
-    url = url_prefix + 'dv/?format=rdb&sites=' + site + '&startDT=' + date + '&endDT=' + date + '&siteStatus=all'
-    try:
-        r = requests.get(url)
-        if r.status_code is 200:
-            content = r.text.splitlines()
-    except requests.exceptions.RequestException as e:
-        print('- Bad URL -')
-    return content
+    Get peak streamflow for a particular day from the daily values service
+    :param site: Site number on which to get data
+    :param date: Date for which to get the peak flow
+    :param url_prefix: NWIS service endpoint
+    :return: List of dictionaries containing the date and value of peak streamflow
+    """
+    url_prefix += 'dv/'
+    params = {
+        'format': 'rdb',
+        'siteStatus': 'all',
+        'sites': site,
+        'startDT': date,
+        'endDT': date
 
+    }
+    content = utils.parse_rdb(url_prefix, params)
+    if content is None:
+        print('rdb parser has returned no data to peakflow util req_peak_dv_data')
+        peaks = None
 
+    else:
+        # Find the name of that column whose name changes
+        for k in content[0].keys():
+            colname = re.match(r'\d+_00060_00003$', k)
+            # if a match has been found
+            if colname is not None:
+                colname = colname.string
+                break
+        # (no break) means that this column has not been found
+        else:
```

Comment:
Is this just a last ditch effort to get the value based on position heuristic if re doesn't work?